src/_pytest/fixtures.py

```diff
@@ -786,13 +786,21 @@ def fail_fixturefunc(fixturefunc, msg):
 def call_fixture_func(fixturefunc, request, kwargs):
     yieldctx = is_generator(fixturefunc)
     if yieldctx:
-        it = fixturefunc(**kwargs)
-        res = next(it)
-        finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, it)
-        request.addfinalizer(finalizer)
+        generator = fixturefunc(**kwargs)
+        try:
+            fixture_return_value = next(generator)
+        except StopIteration:
+            raise ValueError(
+                "Fixture {} did not yield a value".format(fixturefunc.__name__)
+            )
```

Comment:
I don't think the cause is interesting in this case, and just makes the traceback less tidy, so I'd omit it:

```suggestion
            ) from None
```

Reply:
are we sure we don't want to say something? the 7061 issue suggests we should explain the reason to them, this might still be a little vague?

Reply:
I think "Yielding" in "Yielding fixture" is redundant because the continuation is "failed to yield ...".

I also liked that in the previous version the the fixture name was provided in the error message, to pinpoint the issue. BTW, if you bring it back, should probably use `request.fixturename` (the fixture name, can be set explicitly with `@pytest.fixture(name='...')) rather than the name of the fixture function itself.

Finally, I prefer "did not yield a value" over "did not yield a single value", because the latter implies it either didn't yield or yielded more than once, but we don't check for more than once.

Reply:
Hey great feedback thanks! i was wondering how to get the name easier when name='' was provided, great idea! could we agree on the following error message:

`Fixture: {request.fixturename} did not yield a value` ? I will defer to your expertise and implement

Reply:
The ":" makes it read like not a proper sentence, so I prefer without it, unless there is some reason to have it?

Reply:
nope not at all, a bad habbit of mine :) will implement w/o